### PR TITLE
Set statuses on workflows

### PR DESCRIPTION
**Pull request by thomtrp**
Add listener to keep status on workflows up to date:
- version draft => statuses should contain draft
- version active => statuses should contain active
- version deactivated => if no version active, statuses should contain deactivated

Renaming also the endpoints because it was not reflecting the full behaviour.

Finally, adding a new status Archived for versions. Will be used when a version is deactivated, but is not the last published version anymore. It means this version cannot be re-activated.

### CODE REVIEW

File: packages/twenty-server/src/modules/workflow/common/standard-objects/workflow.workspace-entity.ts [hunk 1]
```diff
@@ -18,11 +18,34 @@ import { STANDARD_OBJECT_IDS } from 'src/engine/workspace-manager/workspace-sync
...
+    position: 1,
+    color: 'green',
+  },
+  {
+    value: WorkflowStatus.DEACTIVATED,
+    label: 'Deactivated',
+    position: 2,
+    color: 'grey',
+  },
+];
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/common/standard-objects/workflow.workspace-entity.ts:48]: I think we should add a json constant that stores the valid status combinations:
```
export const validStatuses = {
draftActive: [WorkflowStatus.DRAF, WorkflowStatus.ACTIVE], 
draftDeactivated: [WorkflowStatus.DRAF, WorkflowStatus.DEACTIVATED], 
draft: [WorkflowStatus.DRAF], 
active: [WorkflowStatus.ACTIVE], 
deactivated [WorkflowStatus.DEACTIVATED]
}
```
Then we will need a utils that returns the validStatuses from statuses `export const getValidStatuses(statuses): validStatuses` that we can use in `workflow-update-job.ts`

**Review by martmull — CHANGES_REQUESTED** [review]
Think we should simplify the workflow statuses updates, it will be difficult to maintain. Wrote a suggestion

File: packages/twenty-server/src/modules/workflow/common/standard-objects/workflow.workspace-entity.ts (hunk 1 above)
Comment (thomtrp) [reply packages/twenty-server/src/modules/workflow/common/standard-objects/workflow.workspace-entity.ts:48]: @martmull great feedback! I updated the code

File: packages/twenty-server/src/modules/workflow/workflow-status/utils/workflow-status-combination.util.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+export const getWorkflowStatusCombinationFromArray = (
+  statuses: WorkflowStatus[],
+): WorkflowStatusCombination => {
+  if (
+    statuses.includes(WorkflowStatus.ACTIVE) &&
+    statuses.includes(WorkflowStatus.DRAFT)
+  ) {
+    return WorkflowStatusCombination.ACTIVE_AND_DRAFT;
+  }
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-status/utils/workflow-status-combination.util.ts:21]: I would be more assertive here:
```
if (
statuses === [WorkflowStatus.ACTIVE, WorkflowStatus.DRAFT]
  ) {
    return WorkflowStatusCombination.ACTIVE_AND_DRAFT;
  }
...
```
Would be even nicer to create constants to store combinations:
```
const ACTIVE_AND_DRAFT_COMBINATION = [WorkflowStatus.ACTIVE, WorkflowStatus.DRAFT]
...
```
in order to use them in `getWorkflowStatusesFromCombination`

**Review by martmull — APPROVED** [review]
Very nice

**Pull request merged**